lib/facter/powershell_version.rb

```diff
@@ -0,0 +1,45 @@
+Facter.add(:powershell_version) do
+  setcode do
+    if Puppet::Util::Platform.windows?
+      require 'win32/registry'
+      version = nil
+      begin
+        access_type = Win32::Registry::KEY_READ | 0x100
+        hive        = Win32::Registry::HKEY_LOCAL_MACHINE
+        hive.open('SOFTWARE\Microsoft\PowerShell\3\PowerShellEngine', access_type) do |reg|
+          version = reg['PowerShellVersion']
+        end
+      rescue
+        version = nil
```

Comment:
As [mentioned before](https://github.com/puppetlabs/puppetlabs-dsc/pull/99#discussion_r40114588), this check may need to have a fallback to PowerShell\1\PowerShellEngine when the first call fails. Then maybe another fallback in case PowerShell is not installed at all. Although I'm not really sure how often we'd hit the third scenario given we are moving into places where POSH is already installed by default.